Assertion failure when using bulk hpx::new_ in asynchronous mode

I stumbled upon a weird assertion failure in `hpx/runtime/components/default_distribution_policy.hpp`.
You can reproduce it by adding the following lines to `tests/regressions/components/new_2848.cpp`
```c++
    {
    auto locs = hpx::find_all_localities();
    
    std::vector<hpx::id_type> ids =
        hpx::new_<test_server[]>(hpx::default_layout(locs), 10, 42).get();

    hpx::future<std::vector<hpx::id_type>> ids_f =
        hpx::new_<test_server[]>(hpx::default_layout(locs), 10, 42);
    }
```
and running `new_2848_test` on more than one locality.
```
new_2848_test: /home/igor/hpx/hpx.git/hpx/runtime/components/default_distribution_policy.hpp:163: hpx::components::default_distribution_policy::bulk_create(std::size_t, Ts&& ...) const::<lambda(std::vector<hpx::lcos::future<std::vector<hpx::naming::id_type> > >&&)> mutable [with Component = test_server; Ts = {int}]: Assertion `localities_.size() == v.size()' failed.
```
The weird part here is that `ids` are returned normally, and only the second call to `hpx::new_` crashes.

@hkaiser No problem, it's holiday season after all.

Clang 5.0 does not want to compile #3087...
```
In file included from /home/igor/hpx/hpx.git/hpx/components/containers/unordered/unordered_map.hpp:26:0,
                 from /home/igor/hpx/hpx.git/src/components/containers/unordered/partition_unordered_map_component.cpp:15:
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp: In member function ‘hpx::container_distribution_policy hpx::container_distribution_policy::operator()(std::size_t) const’:
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:38:77: error: no matching function for call to ‘hpx::container_distribution_policy::container_distribution_policy(std::size_t&, const std::shared_ptr<std::vector<hpx::naming::id_type> >&)’
             return container_distribution_policy(num_partitions, localities_);
                                                                             ^
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:114:9: note: candidate: hpx::container_distribution_policy::container_distribution_policy(const hpx::naming::id_type&)
         container_distribution_policy(hpx::id_type const& locality)
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:114:9: note:   candidate expects 1 argument, 2 provided
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:108:9: note: candidate: hpx::container_distribution_policy::container_distribution_policy(std::size_t, std::vector<hpx::naming::id_type>&&)
         container_distribution_policy(std::size_t num_partitions,
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:108:9: note:   no known conversion for argument 2 from ‘const std::shared_ptr<std::vector<hpx::naming::id_type> >’ to ‘std::vector<hpx::naming::id_type>&&’
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:101:9: note: candidate: hpx::container_distribution_policy::container_distribution_policy(std::size_t, const std::vector<hpx::naming::id_type>&)
         container_distribution_policy(std::size_t num_partitions,
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:101:9: note:   no known conversion for argument 2 from ‘const std::shared_ptr<std::vector<hpx::naming::id_type> >’ to ‘const std::vector<hpx::naming::id_type>&’
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:32:9: note: candidate: hpx::container_distribution_policy::container_distribution_policy()
         container_distribution_policy()
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:32:9: note:   candidate expects 0 arguments, 2 provided
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:28:12: note: candidate: hpx::container_distribution_policy::container_distribution_policy(const hpx::container_distribution_policy&)
     struct container_distribution_policy
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:28:12: note:   candidate expects 1 argument, 2 provided
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:28:12: note: candidate: hpx::container_distribution_policy::container_distribution_policy(hpx::container_distribution_policy&&)
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:28:12: note:   candidate expects 1 argument, 2 provided
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp: In member function ‘std::size_t hpx::container_distribution_policy::get_num_partitions() const’:
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:83:29: error: ‘const class std::shared_ptr<std::vector<hpx::naming::id_type> >’ has no member named ‘size’
                 localities_.size() : num_partitions_;
                             ^~~~
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp: In member function ‘const std::vector<hpx::naming::id_type>& hpx::container_distribution_policy::get_localities() const’:
/home/igor/hpx/hpx.git/hpx/components/containers/container_distribution_policy.hpp:89:20: error: invalid initialization of reference of type ‘const std::vector<hpx::naming::id_type>&’ from expression of type ‘const std::shared_ptr<std::vector<hpx::naming::id_type> >’
             return localities_;
                    ^~~~~~~~~~~
```

--- a/hpx/components/containers/container_distribution_policy.hpp
+++ b/hpx/components/containers/container_distribution_policy.hpp
@@ -14,6 +14,7 @@
 #include <hpx/runtime/serialization/serialize.hpp>
 #include <hpx/runtime/serialization/shared_ptr.hpp>
 #include <hpx/runtime/serialization/vector.hpp>
+#include <hpx/util/assert.hpp>
 
 #include <algorithm>
 #include <cstddef>
@@ -30,21 +31,20 @@ namespace hpx
     struct container_distribution_policy
       : components::default_distribution_policy
     {
-    private:
-        typedef components::default_distribution_policy base_type;
-
     public:
         container_distribution_policy()
           : num_partitions_(std::size_t(-1))
         {}
 
-        container_distribution_policy operator()(std::size_t num_partitions) const
+        container_distribution_policy operator()(
+            std::size_t num_partitions) const
         {
-            auto localities = this->base_type::ensure_localities();
-            return container_distribution_policy(num_partitions, *localities);
+            return container_distribution_policy(
+                num_partitions, get_localities());
         }
 
-        container_distribution_policy operator()(hpx::id_type const& locality) const
+        container_distribution_policy operator()(
+            hpx::id_type const& locality) const
         {
             return container_distribution_policy(locality);
         }
@@ -53,7 +53,10 @@ namespace hpx
             std::vector<id_type> const& localities) const
         {
             if (num_partitions_ != std::size_t(-1))
-                return container_distribution_policy(num_partitions_, localities);
+            {
+                return container_distribution_policy(
+                    num_partitions_, localities);
+            }
             return container_distribution_policy(localities.size(), localities);
         }
 
@@ -85,15 +88,25 @@ namespace hpx
         ///////////////////////////////////////////////////////////////////////
         std::size_t get_num_partitions() const
         {
-            auto localities = this->base_type::ensure_localities();
-            std::size_t num_parts = (num_partitions_ == std::size_t(-1)) ?
-                localities->size() : num_partitions_;
-            return (std::max)(num_parts, std::size_t(1));
+            if (localities_)
+            {
+                std::size_t num_parts = (num_partitions_ == std::size_t(-1)) ?
+                    localities_->size() : num_partitions_;
+                return (std::max)(num_parts, std::size_t(1));
+            }
+            return std::size_t(1);
         }
 
-        std::vector<hpx::id_type> const& get_localities() const
+        std::vector<hpx::id_type> get_localities() const
         {
-            return *this->base_type::ensure_localities();
+            if (!localities_)
+            {
+                // use this locality, if this object was default constructed
+                return std::vector<id_type>(1, hpx::find_here());
+            }
+
+            HPX_ASSERT(!localities_->empty());
+            return *localities_;
         }
 
     private:
@@ -110,7 +123,6 @@ namespace hpx
           : components::default_distribution_policy(localities),
             num_partitions_(num_partitions)
         {}
-
 
         container_distribution_policy(std::size_t num_partitions,
                 std::vector<id_type> && localities)

--- a/hpx/components/containers/container_distribution_policy.hpp
+++ b/hpx/components/containers/container_distribution_policy.hpp
@@ -1,5 +1,5 @@
 //  Copyright (c) 2014 Bibek Ghimire
-//  Copyright (c) 2014-2015 Hartmut Kaiser
+//  Copyright (c) 2014-2017 Hartmut Kaiser
 //
 //  Distributed under the Boost Software License, Version 1.0. (See accompanying
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
@@ -12,10 +12,12 @@
 #include <hpx/traits/is_distribution_policy.hpp>
 
 #include <hpx/runtime/serialization/serialize.hpp>
+#include <hpx/runtime/serialization/shared_ptr.hpp>
 #include <hpx/runtime/serialization/vector.hpp>
 
 #include <algorithm>
 #include <cstddef>
+#include <memory>
 #include <type_traits>
 #include <utility>
 #include <vector>
@@ -28,6 +30,9 @@ namespace hpx
     struct container_distribution_policy
       : components::default_distribution_policy
     {
+    private:
+        typedef components::default_distribution_policy base_type;
+
     public:
         container_distribution_policy()
           : num_partitions_(std::size_t(-1))
@@ -35,7 +40,8 @@ namespace hpx
 
         container_distribution_policy operator()(std::size_t num_partitions) const
         {
-            return container_distribution_policy(num_partitions, localities_);
+            auto localities = this->base_type::ensure_localities();
+            return container_distribution_policy(num_partitions, *localities);
         }
 
         container_distribution_policy operator()(hpx::id_type const& locality) const
@@ -79,14 +85,15 @@ namespace hpx
         ///////////////////////////////////////////////////////////////////////
         std::size_t get_num_partitions() const
         {
+            auto localities = this->base_type::ensure_localities();
             std::size_t num_parts = (num_partitions_ == std::size_t(-1)) ?
-                localities_.size() : num_partitions_;
+                localities->size() : num_partitions_;
             return (std::max)(num_parts, std::size_t(1));
         }
 
         std::vector<hpx::id_type> const& get_localities() const
         {
-            return localities_;
+            return *this->base_type::ensure_localities();
         }
 
     private:

--- a/hpx/runtime/components/default_distribution_policy.hpp
+++ b/hpx/runtime/components/default_distribution_policy.hpp
@@ -20,6 +20,7 @@
 #include <hpx/runtime/naming/name.hpp>
 #include <hpx/runtime/serialization/serialization_fwd.hpp>
 #include <hpx/runtime/serialization/vector.hpp>
+#include <hpx/runtime/serialization/shared_ptr.hpp>
 #include <hpx/traits/extract_action.hpp>
 #include <hpx/traits/is_distribution_policy.hpp>
 #include <hpx/traits/promise_local_result.hpp>
@@ -27,6 +28,7 @@
 
 #include <algorithm>
 #include <cstddef>
+#include <memory>
 #include <type_traits>
 #include <utility>
 #include <vector>
@@ -105,7 +107,8 @@ namespace hpx { namespace components
         {
             using components::stub_base;
 
-            for (hpx::id_type const& loc: localities_)
+            auto localities = ensure_localities();
+            for (hpx::id_type const& loc: *localities)
             {
                 if (get_num_items(1, loc) != 0)
                 {
@@ -144,12 +147,13 @@ namespace hpx { namespace components
         {
             using components::stub_base;
 
-            if (localities_.size() > 1)
+            auto localities = ensure_localities();
+            if (localities->size() > 1)
             {
                 // schedule creation of all objects across given localities
                 std::vector<hpx::future<std::vector<hpx::id_type> > > objs;
-                objs.reserve(localities_.size());
-                for (hpx::id_type const& loc: localities_)
+                objs.reserve(localities->size());
+                for (hpx::id_type const& loc: *localities)
                 {
                     objs.push_back(stub_base<Component>::bulk_create_async(
                         loc, get_num_items(count, loc), vs...));
@@ -157,19 +161,18 @@ namespace hpx { namespace components
 
                 // consolidate all results
                 return hpx::dataflow(hpx::launch::sync,
-                    [=](std::vector<hpx::future<std::vector<hpx::id_type> > > && v)
-                        mutable -> std::vector<bulk_locality_result>
+                    [localities](
+                        std::vector<hpx::future<std::vector<hpx::id_type> > > && v
+                    ) mutable -> std::vector<bulk_locality_result>
                     {
-                        HPX_ASSERT(localities_.size() == v.size());
+                        HPX_ASSERT(localities->size() == v.size());
 
                         std::vector<bulk_locality_result> result;
                         result.reserve(v.size());
 
                         for (std::size_t i = 0; i != v.size(); ++i)
                         {
-                            result.emplace_back(
-                                    std::move(localities_[i]), v[i].get()
-                                );
+                            result.emplace_back((*localities)[i], v[i].get());
                         }
                         return result;
                     },
@@ -271,63 +274,85 @@ namespace hpx { namespace components
         ///
         std::size_t get_num_localities() const
         {
-            return (std::max)(std::size_t(1), localities_.size());
+            return (std::max)(std::size_t(1), localities_->size());
         }
 
         /// Returns the locality which is anticipated to be used for the next
         /// async operation
         hpx::id_type get_next_target() const
         {
-            return localities_.empty() ? hpx::find_here() : localities_.front();
+            return (!localities_ || localities_->empty()) ?
+                hpx::find_here() : localities_->front();
         }
 
     protected:
         /// \cond NOINTERNAL
-        std::size_t
-        get_num_items(std::size_t items, hpx::id_type const& loc) const
+        std::size_t get_num_items(
+            std::size_t items, hpx::id_type const& loc) const
         {
             // make sure the given id is known to this distribution policy
             HPX_ASSERT(
-                std::find(localities_.begin(), localities_.end(), loc) !=
-                    localities_.end() ||
-                (localities_.empty() && loc == hpx::find_here())
+               !localities_ ||
+                (localities_->empty() && loc == hpx::find_here()) ||
+                std::find(localities_->begin(), localities_->end(), loc) !=
+                    localities_->end()
             );
+
+            if (!localities_ || localities_->empty())
+            {
+                return std::size_t(1);
+            }
 
             // this distribution policy places an equal number of items onto
             // each locality
-            std::size_t locs = (std::max)(std::size_t(1), localities_.size());
+            std::size_t locs = (std::max)(std::size_t(1), localities_->size());
 
             // the overall number of items to create is smaller than the number
             // of localities
             if (items < locs)
             {
-                auto it = std::find(localities_.begin(), localities_.end(), loc);
-                std::size_t num_loc = std::distance(localities_.begin(), it);
+                auto it = std::find(localities_->begin(), localities_->end(), loc);
+                std::size_t num_loc = std::distance(localities_->begin(), it);
                 return (items < num_loc) ? 1 : 0;
             }
 
             // the last locality might get less items
-            if (localities_.size() > 1 && loc == localities_.back())
+            if (localities_->size() > 1 && loc == localities_->back())
+            {
                 return items - detail::round_to_multiple(items, locs, locs-1);
+            }
 
             // otherwise just distribute evenly
             return (items + locs - 1) / locs;
+        }
+
+        std::shared_ptr<std::vector<id_type>> ensure_localities() const
+        {
+            // use this locality, if this object was default constructed
+            auto localities = localities_;
+            if (!localities || localities->empty())
+            {
+                localities = std::make_shared<std::vector<id_type>>();
+                localities->push_back(hpx::find_here());
+            }
+            return localities;
         }
         /// \endcond
 
     protected:
         /// \cond NOINTERNAL
         default_distribution_policy(std::vector<id_type> const& localities)
-          : localities_(localities)
+          : localities_(std::make_shared<std::vector<id_type>>(localities))
         {}
 
         default_distribution_policy(std::vector<id_type> && localities)
-          : localities_(std::move(localities))
+          : localities_(std::make_shared<std::vector<id_type>>(std::move(localities)))
         {}
 
         default_distribution_policy(id_type const& locality)
+          : localities_(std::make_shared<std::vector<id_type>>())
         {
-            localities_.push_back(locality);
+            localities_->push_back(locality);
         }
 
         friend class hpx::serialization::access;
@@ -338,7 +363,8 @@ namespace hpx { namespace components
             ar & localities_;
         }
 
-        std::vector<id_type> localities_;   // localities to create things on
+        // localities to create things on
+        std::shared_ptr<std::vector<id_type>> localities_;
         /// \endcond
     };
 

--- a/hpx/runtime/components/default_distribution_policy.hpp
+++ b/hpx/runtime/components/default_distribution_policy.hpp
@@ -55,8 +55,7 @@ namespace hpx { namespace components
     public:
         /// Default-construct a new instance of a \a default_distribution_policy.
         /// This policy will represent one locality (the local locality).
-        default_distribution_policy()
-        {}
+        default_distribution_policy() = default;
 
         /// Create a new \a default_distribution policy representing the given
         /// set of localities.
@@ -75,7 +74,7 @@ namespace hpx { namespace components
         /// \param locs     [in] The list of localities the new instance should
         ///                 represent
         default_distribution_policy operator()(
-            std::vector<id_type> && locs) const
+            std::vector<id_type>&& locs) const
         {
             return default_distribution_policy(std::move(locs));
         }
@@ -107,13 +106,15 @@ namespace hpx { namespace components
         {
             using components::stub_base;
 
-            auto localities = ensure_localities();
-            for (hpx::id_type const& loc: *localities)
+            if (localities_)
             {
-                if (get_num_items(1, loc) != 0)
+                for (hpx::id_type const& loc: *localities_)
                 {
-                    return stub_base<Component>::create_async(
-                        loc, std::forward<Ts>(vs)...);
+                    if (get_num_items(1, loc) != 0)
+                    {
+                        return stub_base<Component>::create_async(
+                            loc, std::forward<Ts>(vs)...);
+                    }
                 }
             }
 
@@ -147,19 +148,19 @@ namespace hpx { namespace components
         {
             using components::stub_base;
 
-            auto localities = ensure_localities();
-            if (localities->size() > 1)
+            if (localities_ && localities_->size() > 1)
             {
                 // schedule creation of all objects across given localities
                 std::vector<hpx::future<std::vector<hpx::id_type> > > objs;
-                objs.reserve(localities->size());
-                for (hpx::id_type const& loc: *localities)
+                objs.reserve(localities_->size());
+                for (hpx::id_type const& loc: *localities_)
                 {
                     objs.push_back(stub_base<Component>::bulk_create_async(
                         loc, get_num_items(count, loc), vs...));
                 }
 
                 // consolidate all results
+                auto localities = localities_;
                 return hpx::dataflow(hpx::launch::sync,
                     [localities](
                         std::vector<hpx::future<std::vector<hpx::id_type> > > && v
@@ -274,15 +275,14 @@ namespace hpx { namespace components
         ///
         std::size_t get_num_localities() const
         {
-            return (std::max)(std::size_t(1), localities_->size());
+            return !localities_ ? std::size_t(1) : localities_->size();
         }
 
         /// Returns the locality which is anticipated to be used for the next
         /// async operation
         hpx::id_type get_next_target() const
         {
-            return (!localities_ || localities_->empty()) ?
-                hpx::find_here() : localities_->front();
+            return !localities_ ? hpx::find_here() : localities_->front();
         }
 
     protected:
@@ -292,20 +292,14 @@ namespace hpx { namespace components
         {
             // make sure the given id is known to this distribution policy
             HPX_ASSERT(
-               !localities_ ||
-                (localities_->empty() && loc == hpx::find_here()) ||
+                localities_ &&
                 std::find(localities_->begin(), localities_->end(), loc) !=
                     localities_->end()
             );
 
-            if (!localities_ || localities_->empty())
-            {
-                return std::size_t(1);
-            }
-
             // this distribution policy places an equal number of items onto
             // each locality
-            std::size_t locs = (std::max)(std::size_t(1), localities_->size());
+            std::size_t locs = localities_->size();
 
             // the overall number of items to create is smaller than the number
             // of localities
@@ -317,7 +311,7 @@ namespace hpx { namespace components
             }
 
             // the last locality might get less items
-            if (localities_->size() > 1 && loc == localities_->back())
+            if (locs > 1 && loc == localities_->back())
             {
                 return items - detail::round_to_multiple(items, locs, locs-1);
             }
@@ -325,35 +319,35 @@ namespace hpx { namespace components
             // otherwise just distribute evenly
             return (items + locs - 1) / locs;
         }
-
-        std::shared_ptr<std::vector<id_type>> ensure_localities() const
-        {
-            // use this locality, if this object was default constructed
-            auto localities = localities_;
-            if (!localities || localities->empty())
-            {
-                localities = std::make_shared<std::vector<id_type>>();
-                localities->push_back(hpx::find_here());
-            }
-            return localities;
-        }
         /// \endcond
 
     protected:
         /// \cond NOINTERNAL
         default_distribution_policy(std::vector<id_type> const& localities)
           : localities_(std::make_shared<std::vector<id_type>>(localities))
-        {}
+        {
+            if (localities_->empty())
+            {
+                HPX_THROW_EXCEPTION(invalid_status,
+                    "default_distribution_policy::default_distribution_policy",
+                    "unexpectedly empty list of localities");
+            }
+        }
 
         default_distribution_policy(std::vector<id_type> && localities)
           : localities_(std::make_shared<std::vector<id_type>>(std::move(localities)))
-        {}
+        {
+            if (localities_->empty())
+            {
+                HPX_THROW_EXCEPTION(invalid_status,
+                    "default_distribution_policy::default_distribution_policy",
+                    "unexpectedly empty list of localities");
+            }
+        }
 
         default_distribution_policy(id_type const& locality)
-          : localities_(std::make_shared<std::vector<id_type>>())
-        {
-            localities_->push_back(locality);
-        }
+          : localities_(std::make_shared<std::vector<id_type>>(1, locality))
+        {}
 
         friend class hpx::serialization::access;
 

--- a/tests/regressions/components/CMakeLists.txt
+++ b/tests/regressions/components/CMakeLists.txt
@@ -1,9 +1,10 @@
-# Copyright (c) 2007-2016 Hartmut Kaiser
+# Copyright (c) 2007-2017 Hartmut Kaiser
 #
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 set(tests
+    bulk_new_3054
     client_base_registration
     create_n_components_2323
     create_remote_component_2334
@@ -13,6 +14,7 @@ set(tests
     returned_client_2150
    )
 
+set(bulk_new_3054_PARAMETERS LOCALITIES 2)
 set(create_remote_component_2334_PARAMETERS LOCALITIES 2)
 set(new_2848_PARAMETERS LOCALITIES 2)
 

--- a/tests/regressions/components/bulk_new_3054.cpp
+++ b/tests/regressions/components/bulk_new_3054.cpp
@@ -1,0 +1,39 @@
+//  Copyright (c) 2017 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_main.hpp>
+#include <hpx/include/components.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+struct test_server : hpx::components::component_base<test_server>
+{
+    test_server(int) {}
+};
+
+typedef hpx::components::component<test_server> server_type;
+HPX_REGISTER_COMPONENT(server_type, test_server);
+
+void test_bulk_new()
+{
+    auto locs = hpx::find_all_localities();
+
+    std::vector<hpx::id_type> ids =
+        hpx::new_<test_server[]>(hpx::default_layout(locs), 10, 42).get();
+
+    hpx::future<std::vector<hpx::id_type>> ids_f =
+        hpx::new_<test_server[]>(hpx::default_layout(locs), 10, 42);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int main()
+{
+    test_bulk_new();
+
+    return 0;
+}
+

